tests/utapi/utilities.js

```diff
@@ -0,0 +1,282 @@
+    action: 'getBucketReplication',
+    metrics: {
+        bucket: 'source-bucket',
+    },
+}, {
+    action: 'getBucketVersioning',
+    metrics: {
+        bucket: 'bucket-with-object-lock',
+    },
+}, {
+    action: 'getBucketWebsite',
+    metrics: {
+        bucket: 'testgetwebsitetestbucket',
+    },
+}, {
+    action: 'getObjectTagging',
+    metrics: {
+        bucket: 'completempu1615102906771',
+        keys: ['keywithtags'],
+        versionId: undefined,
+        location: 'us-east-1',
+    },
+}, {
+    action: 'headObject',
+    metrics: {
+        bucket: 'supersourcebucket81033016532',
+        keys: ['supersourceobject'],
+        versionId: undefined,
+        location: 'us-east-1',
+    },
+}, {
+    action: 'initiateMultipartUpload',
+    metrics: {
+        bucket: 'destinationbucket815502017',
+        keys: ['copycatobject'],
+        location: 'us-east-1',
+    },
+}, {
+    action: 'listMultipartUploadParts',
+    metrics: {
+        bucket: 'ftest-mybucket-74',
+    },
+}, {
+    action: 'multiObjectDelete',
+    metrics: {
+        bucket: 'completempu1615102906771',
+        keys: [undefined],
+        byteLength: 3,
+        numberOfObjects: 1,
+        isDelete: true,
+    },
+}, {
+    action: 'putBucketAcl',
+    metrics: {
+        bucket: 'putbucketaclfttest',
+    },
+}, {
+    action: 'putBucketCors',
+    metrics: {
+        bucket: 'testcorsbucket',
+    },
+}, {
+    action: 'putBucketNotification',
+    metrics: {
+        bucket: 'notificationtestbucket',
+    },
+}, {
+    action: 'putBucketObjectLock',
+    metrics: {
+        bucket: 'mock-bucket',
+    },
+}, {
+    action: 'putBucketReplication',
+    metrics: {
+        bucket: 'source-bucket',
+    },
+}, {
+    action: 'putBucketVersioning',
+    metrics: {
+        bucket: 'source-bucket',
+    },
+}, {
+    action: 'putBucketWebsite',
+    metrics: {
+        bucket: 'testgetwebsitetestbucket',
+    },
+}, {
+    action: 'uploadPart',
+    metrics: {
+        bucket: 'ftest-mybucket-74',
+        keys: ['toAbort&<>"\''],
+        newByteLength: 5242880,
+        oldByteLength: null,
+        location: 'us-east-1',
+    },
+}, {
+    action: 'uploadPartCopy',
+    metrics: {
+        bucket: 'destinationbucket815502017',
+        keys: ['copycatobject'],
+        newByteLength: 26,
+        oldByteLength: null,
+        location: 'us-east-1',
+    },
+}];
+
+function computeMetrics(action, event) {
+    const {
+        numberOfObjects,
+        oldByteLength,
+        newByteLength,
+        byteLength,
+        isDelete,
+    } = event;
+    const incomingBytes = action === 'getObject' ? 0 : newByteLength;
+    const sizeDelta = oldByteLength ? newByteLength - oldByteLength :
+        incomingBytes;
+    return {
+        objectDelta: isDelete ? -numberOfObjects : numberOfObjects,
+        sizeDelta: isDelete ? -byteLength : sizeDelta,
+        incomingBytes,
+        outgoingBytes: action === 'getObject' ? newByteLength : 0,
+    };
+}
```

Comment:
Duplicating the logic into the test as the code itself isn't the best method of writing tests. For one it ties our tests to the internal logic which we should not do, we just want to test the observable result (eg the output). Additionally, this just proves that what comes in matches what comes out and not that what comes out is correct. For example, if you wrote the test by copy pasting the broken code, the test would still pass even though it was wrong.

Instead, create `expected` objects to compare to in your test array.

Basically it would be the same as doing: 
```js
const jsonString = '{"a": 5}';
assert.strictEqual(JSON.stringify(JSON.parse(jsonString)), jsonString);
```

Reply:
I am still working on this. This is not quite up for review yet, will convert this to draft. I have the correction but did not push it yet... too soon @miniscruff :)